File: Examples/VisualBasic/Loading-and-Saving/ExtractAll3DScenes.vb
Imports System.IO
Imports System.Text
Imports System.Collections.Generic
Imports Aspose.ThreeD
Imports Aspose.ThreeD.Formats

Namespace Loading_Saving
    Class ExtractAll3DScenes
        Public Shared Sub Run()
            'ExStart:ExtractAll3DScenes
            ' The path to the documents directory.
            Dim MyDir As String = RunExamples.GetDataDir()

            Dim password As Byte() = Nothing
            Dim scenes As List(Of Scene) = FileFormat.PDF.ExtractScene(MyDir & Convert.ToString("House_Design.pdf"), password)
            Dim i As Integer = 1
            ' Iterate through the scenes and save in 3D files
            For Each scene As Scene In scenes
                Dim fileName As String = "3d-" + (System.Math.Max(System.Threading.Interlocked.Increment(i), i - 1)) + ".fbx"
                scene.Save(fileName, FileFormat.FBX7400ASCII)
            Next
            'ExEnd:ExtractAll3DScenes            
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Loading-and-Saving/ExtractAll3DScenes.cs
using System;
using System.IO;
using System.Text;
using System.Collections.Generic;
using Aspose.ThreeD;
using Aspose.ThreeD.Formats;

namespace Aspose._3D.Examples.CSharp.Loading_Saving
{
    class ExtractAll3DScenes
    {
        public static void Run()
        {
            // ExStart:ExtractAll3DScenes

            byte[] password = null;
            List<Scene> scenes = FileFormat.PDF.ExtractScene(RunExamples.GetDataFilePath("House_Design.pdf"), password);
            int i = 1;
            // Iterate through the scenes and save in 3D files
            foreach (Scene scene in scenes)
            {
                string fileName = "3d-" + (i++) + ".fbx";
                scene.Save(RunExamples.GetOutputFilePath(fileName), FileFormat.FBX7400ASCII);
            }
            // ExEnd:ExtractAll3DScenes            
        }
    }
}
